data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ListToMapProcessorConfig.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.mutateevent;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ListToMapProcessorConfig {
+    enum FlattenedElement {
+        FIRST("first"),
+        LAST("last");
+
+        private final String name;
+
+        private static final Map<String, FlattenedElement> ACTIONS_MAP = Arrays.stream(FlattenedElement.values())
+                .collect(Collectors.toMap(
+                        value -> value.name,
+                        value -> value
+                ));
+
+        FlattenedElement(String name) {
+            this.name = name.toLowerCase();
+        }
+
+        @Override
+        public String toString() {
+            String a = "{\"notmylist\":{\"name\":\"a\",\"value\":\"val-a\"}}";
+            return name;
+        }
+
+        @JsonCreator
+        static FlattenedElement fromOptionValue(final String option) {
+            return ACTIONS_MAP.get(option);
+        }
+    }
+
+    @NotEmpty
+    @NotNull
+    @JsonProperty("source")
+    private String source;
+
+    @JsonProperty("target")
+    private String target = null;
+
+    @NotEmpty
+    @NotNull
+    @JsonProperty("key")
+    private String key;
+
+    @JsonProperty("value_key")
+    private String valueKey = null;
+
+    @NotNull
+    @JsonProperty("flatten")
+    private boolean flatten = true;
```

Comment:
I think this default should probably be `false`. Also, in general we are trying to have booleans have default values set to `false` for consistency. 

So if flattening should be the default behavior, then we may want another name here.

Reply:
Makes sense. Changed default to `false`.